src/server/list_family.cc

```diff
@@ -1139,13 +1197,16 @@ void ListFamily::BPopGeneric(ListDir dir, CmdArgList args, ConnectionContext* cn
   VLOG(1) << "BPop timeout(" << timeout << ")";
 
   Transaction* transaction = cntx->transaction;
-  std::string popped_key;
-  std::string popped_value;
+
+  absl::StrAppend(debugMessages.Next(), "BPopGeneric by ", transaction->DebugId());
```

Comment:
hmm, I suggest that you constrain this to shard threads only because you print only the threadlocal log anyway.
so any pieces belonging to the same transaction but added in I/O thread will be likely lost.